javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/matcher/ReferenceMatcher.java

```diff
@@ -87,6 +94,10 @@ private boolean doesMatch(ClassLoader loader) {
   }
 
   private static TypePool createTypePool(ClassLoader loader) {
+    Objects.requireNonNull(loader); // "proxy bootstrap" must be passed instead of null
+
+    // it's ok to use locationStrategy() without fallback bootstrap proxy here since the class
+    // loader cannot be null
     return AgentTooling.poolStrategy()
         .typePool(AgentTooling.locationStrategy().classFileLocator(loader), loader);
```

Comment:
this took me some time to convince myself that it was ok to use locationStrategy without  fallback bootstrap proxy here, so thought these comments / `requireNonNull` validations may help others (this module would definitely benefit from `nullaway`, then we wouldn't need these `requireNonNull` -- hm, maybe I'll turn these `requireNonNull` just into method comment

Reply:
I actually like `Objects.requireNonNull` in general. As long as this will not end in customer JVM crash :)